fontc/src/config.rs

```diff
@@ -39,32 +39,25 @@ impl Config {
     ///
     /// This serializes the configuration if necessary, and returns any existing
     /// incremental state for this run if it exists.
-    //FIXME: this returns a Result but it looks like we never actually return
-    //an error?
-    pub fn init(&self) -> Result<Input, io::Error> {
+    pub fn init(&self) -> Result<Input, Error> {
         let config_file = self.file();
 
         let ir_paths = IrPaths::new(&self.args.build_dir);
         let ir_input_file = ir_paths.ir_input_file();
         if self.has_changed(&config_file) {
             log::info!("Config changed, generating a new one");
             if ir_input_file.exists() {
-                fs::remove_file(ir_input_file).expect("Unable to delete old ir input file");
+                fs::remove_file(ir_input_file)?;
             }
-            fs::write(
-                config_file,
-                serde_yaml::to_string(self).expect("Unable to make yaml for config"),
-            )
-            .expect("Unable to write updated config");
+            fs::write(config_file, serde_yaml::to_string(self)?)?;
         };
 
-        let ir_input = if ir_input_file.exists() {
-            let yml = fs::read_to_string(ir_input_file).expect("Unable to load ir input file");
-            serde_yaml::from_str(&yml).expect("Unable to parse ir input file")
+        if ir_input_file.exists() {
+            let yml = fs::read_to_string(ir_input_file)?;
+            serde_yaml::from_str(&yml).map_err(Into::into)
         } else {
-            Input::new()
-        };
-        Ok(ir_input)
+            Ok(Input::new())
+        }
```

Comment:
Optional: reading this again I dislike the nesting in both before and after. Guessing this could be converted to a chain, e.g. input_file.exists().then(...).or_else.